src/Elastic.Apm/Model/Span.cs

```diff
@@ -47,14 +48,22 @@ internal class Span : ISpan
 			_enclosingTransaction = enclosingTransaction;
 			Name = name;
 			Type = type;
-			IsSampled = isSampled;
 
 			ParentId = parentId;
 			TraceId = traceId;
 
 			if (IsSampled)
-				// Started spans should be counted only for sampled transactions
-				enclosingTransaction.SpanCount.IncrementStarted();
+			{
+				// Started and dropped spans should be counted only for sampled transactions
+				if (enclosingTransaction.SpanCount.IncrementTotal() > _configurationReader.TransactionMaxSpans
```

Comment:
Since you were concerned about multi-thread safety with this new feature you might consider adding a test with multiple threads but only if you think it's worth your time. In case you decide that it's worth the time there is a test helper utility `MultiThreadsTestUtils.TestOnThreads` you can use to do all the thread starting, stopping and result gathering. Below is one of its usages in already existing tests:
https://github.com/elastic/apm-agent-dotnet/blob/045a07bfc1d77962b98cca78779ea4be063851c7/test/Elastic.Apm.Tests/HelpersTests/AgentSpinLockTests.cs#L113

Reply:
@SergeyKleyman, I'll look into it in the weekend. If you it's block release or smth else, then you can merge it and tests will be in the next PR.

Reply:
@SergeyKleyman, I added test for concurrent capturing spans. I don't use `MultiThreadsTestUtils.NumberOfThreadsForTest` to be sure concurrent effect can be achieved independently from hardware and test environment.

Reply:
Thank you - it looks good.